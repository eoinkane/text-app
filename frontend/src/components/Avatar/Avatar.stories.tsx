import React from "react";
import User from "../../models/User/";
import AvatarComponent from "./index";

export default {
  title: "Avatar",
  component: AvatarComponent,
};

export const AvatarWithFullUser = () => {
  const user = new User();
  user.init({
    id: 0,
    username: "test",
    firstName: "Jane",
    lastName: "Doe",
  });
  return <AvatarComponent user={user} />;
};
