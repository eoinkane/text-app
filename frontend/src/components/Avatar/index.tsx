import React, { Fragment } from "react";
import User from "../../models/User/";
import { Avatar as MaterialUIAvatar } from "@material-ui/core";

interface IAvatarProps {
  user: User;
}

const Avatar: React.FC<IAvatarProps> = ({ user }) => {
  const firstName = user.get("firstName") as string;
  const lastName = user.get("lastName") as string;
  const initals =
    !firstName || !lastName ? null : `${firstName[0]}${lastName[0]}`;
  return (
    <Fragment>
      <MaterialUIAvatar>{initals}</MaterialUIAvatar>
    </Fragment>
  );
};

export default Avatar;
