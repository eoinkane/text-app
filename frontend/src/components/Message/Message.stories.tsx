import React from "react";
import MessageComponent from "./index";
import Container from "@material-ui/core/Container";

export default {
  title: "Message",
  component: MessageComponent,
};

export const Message = () => {
  const text =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ";
  return (
    <Container style={{ backgroundColor: "#f5f5f5", padding: "4vh 4vw" }}>
      <MessageComponent text={text} />
    </Container>
  );
};
