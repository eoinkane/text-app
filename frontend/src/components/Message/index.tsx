import React from "react";
import cx from "clsx";
import { Paper, Typography } from "@material-ui/core";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

interface MessagePropsBase {}

interface MessagePropsLeft extends MessagePropsBase {
  left: boolean;
}

interface MessagePropsRight extends MessagePropsBase {
  right: boolean;
}

type MessageProps = MessagePropsLeft | MessagePropsRight;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      maxWidth: "85%",
      padding: "10px 0",
    },
    left: {
      marginRight: "auto",
    },
    right: {
      marginLeft: "auto",
    },
    paper: {
      padding: "2vh 2vh",
      borderRadius: 10,
    },
  })
);

const Message: React.FC<MessageProps> = ({ children, ...rest }) => {
  const classes = useStyles();
  return (
    <div
      className={cx(
        classes.root,
        {
          [classes.left]:
            rest.hasOwnProperty("left") && (rest as MessagePropsLeft).left,
        },
        {
          [classes.right]:
            rest.hasOwnProperty("right") && (rest as MessagePropsRight).right,
        }
      )}
    >
      <Paper className={classes.paper} elevation={3}>
        <Typography>{children}</Typography>
      </Paper>
    </div>
  );
};

export default Message;
