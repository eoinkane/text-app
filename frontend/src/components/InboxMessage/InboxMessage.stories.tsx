import React from "react";
import InboxMessageComponent from "./index";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Message from "../../models/Message";
import User from "../../models/User";

export default {
  title: "InboxMessage",
  component: InboxMessageComponent,
};

export const InboxMessage = () => {
  const currentUser = new User();
  currentUser.init({
    username: "test",
    id: 0,
    firstName: "Jane",
    lastName: "Doe",
  });
  const sentMessage = new Message({
    sender: currentUser,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });
  const otherParticipant = new User();
  otherParticipant.init({
    username: "test",
    id: 0,
    firstName: "John",
    lastName: "Appleseed",
  });
  const recievedMessage = new Message({
    sender: otherParticipant,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });
  return (
    <Container style={{ padding: "4vh 4vw" }}>
      <Typography
        style={{
          textAlign: "center",
          paddingBottom: 10,
        }}
      >
        Sent Inbox Message
      </Typography>
      <Container
        style={{
          backgroundColor: "#f5f5f5",
          padding: "20px 20px",
          margin: "20px 0",
        }}
      >
        <InboxMessageComponent
          sent
          otherParticipant={otherParticipant}
          message={sentMessage}
        />
      </Container>

      <Typography
        style={{
          textAlign: "center",
          paddingBottom: 10,
        }}
      >
        Recieved Inbox Message
      </Typography>
      <Container
        style={{
          backgroundColor: "#f5f5f5",
          padding: "20px 20px",
          margin: "20px 0",
        }}
      >
        <InboxMessageComponent
          recieved
          otherParticipant={otherParticipant}
          message={recievedMessage}
        />
      </Container>
    </Container>
  );
};
