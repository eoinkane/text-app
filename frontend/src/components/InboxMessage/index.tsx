import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { Paper, ListItemText } from "@material-ui/core";
import Message from "../../models/Message";
import User from "../../models/User";

interface InboxMessageBase {
  currentUser: User;
  otherParticipant: User;
  message: Message;
}

interface InboxMessageSent extends InboxMessageBase {
  sent: boolean;
}

interface InboxMessageRecieved extends InboxMessageBase {
  recieved: boolean;
}

type InboxMessageProps = InboxMessageSent | InboxMessageRecieved;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    paper: {
      padding: "2vh 2vw",
      borderRadius: 10,
    },
    messageTypography: {
      overflow: "hidden",
      maxHeight: "3vh",
    },
  })
);

const InboxMessage: React.FC<InboxMessageProps> = ({
  currentUser,
  otherParticipant,
  message,
  ...rest
}) => {
  const classes = useStyles();

  const otherParticipantName = `${
    otherParticipant.get("firstName") as string
  } ${otherParticipant.get("lastName") as string}`;

  let prefix;
  if ((rest as Object).hasOwnProperty("recieved")) {
    if ((rest as InboxMessageRecieved).recieved) {
      const firstName = (message.get("sender") as User).get(
        "firstName"
      ) as string;
      const lastName = (message.get("sender") as User).get(
        "lastName"
      ) as string;
      prefix = `${firstName[0]}${lastName[0]}`;
    }
  } else if ((rest as Object).hasOwnProperty("sent")) {
    if ((rest as InboxMessageSent).sent) {
      prefix = `You`;
    }
  }

  const secondaryText = `${prefix}: ${message.get("message") as string}`;

  return (
    <div className={classes.root}>
      <Paper className={classes.paper} elevation={3}>
        <ListItemText
          primary={otherParticipantName}
          secondary={secondaryText}
          secondaryTypographyProps={{
            className: classes.messageTypography,
          }}
        />
      </Paper>
    </div>
  );
};

export default InboxMessage;
