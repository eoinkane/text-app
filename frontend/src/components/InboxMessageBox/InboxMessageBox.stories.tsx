import React, { Fragment } from "react";
import InboxMessageBoxComponent from "./index";
import Message from "../../models/Message";
import User from "../../models/User";
import Typography from "@material-ui/core/Typography";

export default {
  title: "InboxMessageBox",
  component: InboxMessageBoxComponent,
};

const currentUser = new User({ firstName: "Jane", lastName: "Doe" });
const sentMessage = new Message({
  sender: currentUser,
  message:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
});
const otherParticipant = new User({
  firstName: "John",
  lastName: "Appleseed",
});
const recievedMessage = new Message({
  sender: otherParticipant,
  message:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
});

export const InboxMessageBox = () => {
  return (
    <Fragment>
      <Typography style={{ textAlign: "center", paddingBottom: 10 }}>
        Recieved Message
      </Typography>

      <InboxMessageBoxComponent
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={recievedMessage}
      />

      <Typography style={{ textAlign: "center", paddingBottom: 10 }}>
        Sent Message
      </Typography>
      <InboxMessageBoxComponent
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    </Fragment>
  );
};

export const Sent = () => (
  <InboxMessageBoxComponent
    currentUser={currentUser}
    otherParticipant={otherParticipant}
    message={sentMessage}
  />
);

export const Recieved = () => (
  <InboxMessageBoxComponent
    currentUser={currentUser}
    otherParticipant={otherParticipant}
    message={recievedMessage}
  />
);
