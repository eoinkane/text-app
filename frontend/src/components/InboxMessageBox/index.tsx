import React from "react";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import { Grid, Container } from "@material-ui/core";

import Avatar from "./../Avatar/";
import InboxMessage from "./../InboxMessage/";

import User from "../../models/User";
import Message from "../../models/Message";

interface InboxMessageBoxProps {
  currentUser: User;
  otherParticipant: User;
  message: Message;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
    },
    container: {
      backgroundColor: "#f5f5f5",
      maxWidth: "inherit",
      padding: "1vh 0",
    },
    inboxMessageGrid: {
      marginRight: "2vw",
    },
    avatarGrid: {
      flexBasis: 0,
      marginRight: "auto",
      marginLeft: "auto",
      marginBottom: "auto",
      marginTop: "auto",
      paddingRight: "2vw",
      paddingLeft: "2vw",
    },
  })
);

const InboxMessageBox: React.FC<InboxMessageBoxProps> = ({
  currentUser,
  otherParticipant,
  message,
}) => {
  const classes = useStyles();

  const messageSender = message.get("sender") as User;
  const sent = messageSender === currentUser;
  const recieved = messageSender === otherParticipant;

  return (
    <div className={classes.root}>
      <Container className={classes.container}>
        <Grid
          container
          direction="row"
          justify="flex-start"
          alignItems="flex-end"
        >
          <Grid className={classes.avatarGrid} item xs={11}>
            <Avatar user={otherParticipant} />
          </Grid>
          <Grid className={classes.inboxMessageGrid} item xs>
            <InboxMessage
              currentUser={currentUser}
              otherParticipant={otherParticipant}
              sent={sent}
              recieved={recieved}
              message={message}
            />
          </Grid>
        </Grid>
      </Container>
    </div>
  );
};

export default InboxMessageBox;
