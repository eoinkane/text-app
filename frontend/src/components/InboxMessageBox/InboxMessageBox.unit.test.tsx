import React from "react";
import { shallow } from "enzyme";
import InboxMessageBox from "./index";
import Message from "../../models/Message";
import User from "../../models/User";
import Avatar from "../Avatar";
import InboxMessage from "../InboxMessage";
import { Grid } from "@material-ui/core";

describe("InboxMessageBox", () => {
  const currentUser = new User({ firstName: "Jane", lastName: "Doe" });
  const sentMessage = new Message({
    sender: currentUser,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });
  const otherParticipant = new User({
    firstName: "John",
    lastName: "Appleseed",
  });
  const recievedMessage = new Message({
    sender: otherParticipant,
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });

  it("Renders ", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();
  });

  it("Renders a Avatar", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();

    expect(inboxMessageBox.find(Avatar).length).toBe(1);
  });

  it("Renders a InboxMessage", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();

    expect(inboxMessageBox.find(InboxMessage).length).toBe(1);
  });

  it("Renders InboxMessage in sent mode when last message is from current user", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();

    expect(inboxMessageBox.find(InboxMessage).prop("sent")).toBe(true);
    expect(inboxMessageBox.find(InboxMessage).prop("recieved")).toBe(false);
  });

  it("Renders InboxMessage in recieved mode when last message is from otherParticipant", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={recievedMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();

    expect(inboxMessageBox.find(InboxMessage).prop("sent")).toBe(false);
    expect(inboxMessageBox.find(InboxMessage).prop("recieved")).toBe(true);
  });

  it("Passes InboxMessage correct props", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();

    expect(inboxMessageBox.find(InboxMessage).prop("currentUser")).toBe(
      currentUser
    );
    expect(inboxMessageBox.find(InboxMessage).prop("otherParticipant")).toBe(
      otherParticipant
    );
    expect(inboxMessageBox.find(InboxMessage).prop("message")).toBe(
      sentMessage
    );
  });

  it("Passes Avatar correct props", () => {
    const inboxMessageBox = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessageBox).toMatchSnapshot();

    expect(inboxMessageBox.find(Avatar).prop("user")).toBe(otherParticipant);
  });

  it("Applys the correct class for the avatar grid", () => {
    const inboxMessage = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessage).toMatchSnapshot();

    expect(
      inboxMessage.find(Grid).at(1).hasClass("makeStyles-avatarGrid-4")
    ).toBe(true);
  });

  it("Renders the correct padding for InboxMessage grid", () => {
    const inboxMessage = shallow(
      <InboxMessageBox
        currentUser={currentUser}
        otherParticipant={otherParticipant}
        message={sentMessage}
      />
    );
    expect(inboxMessage).toMatchSnapshot();

    expect(
      inboxMessage.find(Grid).at(2).hasClass("makeStyles-inboxMessageGrid-3")
    ).toBe(true);
  });
});
