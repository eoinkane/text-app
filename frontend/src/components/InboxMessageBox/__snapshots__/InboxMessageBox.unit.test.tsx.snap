// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InboxMessageBox Applys the correct class for the avatar grid 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Passes Avatar correct props 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Passes InboxMessage correct props 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Renders  1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Renders InboxMessage in recieved mode when last message is from otherParticipant 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "John",
                    "lastName": "Appleseed",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={true}
          sent={false}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Renders InboxMessage in sent mode when last message is from current user 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Renders a Avatar 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Renders a InboxMessage 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;

exports[`InboxMessageBox Renders the correct padding for InboxMessage grid 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="flex-end"
      container={true}
      direction="row"
      justify="flex-start"
    >
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-avatarGrid-4"
        item={true}
        xs={11}
      >
        <Avatar
          user={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="makeStyles-inboxMessageGrid-3"
        item={true}
        xs={true}
      >
        <InboxMessage
          currentUser={
            User {
              "data": Object {
                "firstName": "Jane",
                "lastName": "Doe",
              },
            }
          }
          message={
            Message {
              "data": Object {
                "message": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
                "sender": User {
                  "data": Object {
                    "firstName": "Jane",
                    "lastName": "Doe",
                  },
                },
              },
            }
          }
          otherParticipant={
            User {
              "data": Object {
                "firstName": "John",
                "lastName": "Appleseed",
              },
            }
          }
          recieved={false}
          sent={true}
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Container))>
</div>
`;
